### PR TITLE
Konveyor and OLM add-ons

### PR DESCRIPTION
**Pull request by freschri**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/konveyor/tackle-manifest.ytpl [hunk 1]
```diff
@@ -0,0 +1,39 @@
+kind: Tackle
+apiVersion: tackle.konveyor.io/v1alpha1
```
Comment (shapirov103) [inline lib/addons/konveyor/tackle-manifest.ytpl:2]: v1alpha1 makes me think this product is not production ready.

File: lib/addons/olm/index.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        OLMDownloadURL+this.props.version+"/olm.yaml"
+      ];
+    }
+
+    deploy(clusterInfo: ClusterInfo): void | Promise<Construct> {
+      let previousResource: Construct;
+      const resources: Construct[] = [];
+      this.manifestUrls!.map((manifestUrl, urlIndex) => {
+        /* eslint-disable */
+        const request = require('sync-request');
```
Comment (shapirov103) [inline lib/addons/olm/index.ts:39]: This loads the sync-request library in a cycle redefining the function. Can you move it out of the loop?

File: lib/addons/olm/index.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        // 2. remove descriptions
+        const doc: string = response.getBody().toString();
+        const trimmedDoc = doc.replace(/description:.*[a-zA-Z0-9].*\n/g, 'description: removed\n');
+        const manifests = trimmedDoc.split("---").filter(e=>e.length>0).map(e => loadYaml(e));
+
+        const batchResources = manifests.map((manifest, docIndex) => {
+          const manifestDeployment: ManifestDeployment = {
+            name: "olmIdx"+urlIndex+docIndex,
+            namespace: "olm",
+            manifest: [manifest],
```
Comment (shapirov103) [inline lib/addons/olm/index.ts:52]: why manifest is an array here? should be an actual doc.

File: lib/addons/olm/index.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            values: {}
+          };
+  
+          const kubectlProvider = new KubectlProvider(clusterInfo);
```
Comment (shapirov103) [inline lib/addons/olm/index.ts:56]: there is no need to instantiate kubectlProvider in a loop. let's move out

File: docs/addons/konveyor.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Konveyor Amazon EKS Blueprints AddOn
+
+This add-on installs [Konveyor](https://konveyor.github.io/).
```
Comment (shapirov103) [inline docs/addons/konveyor.md:3]: Please elaborate on the use cases  answering the question of why the customer would use this addon, expanding on scenarios where this product is applicable.

File: lib/addons/olm/index.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            resource.node.addDependency(previousResource);
+          }
+          previousResource = resource;
+          return resource;
+        });
+
+        resources.push(...batchResources);
+      });
+
+      return Promise.resolve(resources[0]);
```
Comment (shapirov103) [inline lib/addons/olm/index.ts:69]: should not this be last resource as opposed to 0? If it is first, then it won't wait for the rest of the resources to apply.

**Review by shapirov103 — COMMENTED** [review]
@freschri I added some tactical feedback, however on a more strategic items I have a few questions:

1. https://github.com/konveyor/konveyor.github.io does not appear to be production ready. I don't any releases and the versioning schema they are apply is in the alpha range. In addition the repo does not appear to be actively maintained. Most contributions happened last year and I don't observe active development on it. We will only take actively maintained popular OSS projects here, so it appears that this addon may be best served in your personal github repo for now. 
If you have access to the contributors and/or partner sponsoring the OSS then it may be best to move to their repo and we can include some doc updates on it. This PR can be split to just keep the OLM part which is indeed an enabler. 

2. OLM being non-helm addon : how do you envision maintenance of this add-on, e.g. version upgrades and testing against latest EKS versions? Do I have your commitment on ownership of this part?

**Comment by jwmatthews** [comment]
> @freschri I added some tactical feedback, however on a more strategic items I have a few questions:
> 
> 1. https://github.com/konveyor/konveyor.github.io does not appear to be production ready. I don't any releases and the versioning schema they are apply is in the alpha range. In addition the repo does not appear to be actively maintained. Most contributions happened last year and I don't observe active development on it. We will only take actively maintained popular OSS projects here, so it appears that this addon may be best served in your personal github repo for now.

Hello, the repo above is not representative of Konveyor.  Konveyor is actively being developed and maintained. 

That is one of the upstream documentation repos we created to have a single consolidated spot for documentation, we needed to pause that effort and switched to documenting areas in their respective repositories.  We are beginning to get back to the consolidated doc effort, but I wouldn't consider it representative to gauge activity on Konveyor.

Konveyor releases as an Operator, as such the first place to look would be the Operator repo which is at:

- https://github.com/konveyor/tackle2-operator

The Konveyor operator deploys the components below:

- Analysis: https://github.com/windup/windup
- Hub: https://github.com/konveyor/tackle2-hub
- UI: https://github.com/konveyor/tackle2-ui
- Assessments: https://github.com/konveyor/tackle-pathfinder

The majority of development activity in Konveyor is centered on the Hub + UI.

>    If you have access to the contributors and/or partner sponsoring the OSS then it may be best to move to their repo and we can include some doc updates on it. This PR can be split to just keep the OLM part which is indeed an enabler.
> 2. OLM being non-helm addon : how do you envision maintenance of this add-on, e.g. version upgrades and testing against latest EKS versions? Do I have your commitment on ownership of this part?

File: lib/addons/konveyor/tackle-manifest.ytpl (hunk 1 above)
Comment (jwmatthews) [reply lib/addons/konveyor/tackle-manifest.ytpl:2]: Hello,

I'm a maintainer of Konveyor and can provide some more context of Konveyor's status.
Konveyor is an actively developed and maintained CNCF sandbox project that is the upstream for [Red Hat's Migration Toolkit for Applications ](https://developers.redhat.com/products/mta/overview).   

The code is in production at multiple large organizations being used to aid modernization engagements of enterprise java applications to Kubernetes.  The code that makes up Konveyor began ~10 years ago on the Analysis side with the latest version of the platform being developed ~2 years ago, the solution was called Tackle2.  

After being accepted into CNCF,  there has been an effort to refocus Konveyor onto a [Unified vision](https://github.com/konveyor/enhancements/tree/master/enhancements/unified_experience) of bringing the functionality that used to be in separate projects together into a common solution which is Konveyor.  

As part of this refocus, we decided to release Konveyor as a "v0.1" as opposed to continuing with the Tackle 2.1+ scheme that was in place prior.  

It is possible for the API surface to change as we are working with the community to shape what the future will be.  In regard to the API surface for the operator CR, we are careful to keep upgrades working and stable and have automated tests and QE resources actively helping to ensure stability.

File: docs/addons/konveyor.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+## Dependencies
+
+The add-on depends on the following components:
+- Operator Lifecycle Manager add-on (_OlmAddOn_)
+- [AWS Load Balancer Controller](https://kubernetes-sigs.github.io/aws-load-balancer-controller/) add-on (_AwsLoadBalancerControllerAddOn_), which will instantiate an [ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html) to fulfill the Ingress deployed by this add-on
+- [EBS CSI Driver Amazon EKS add-on](https://aws-quickstart.github.io/cdk-eks-blueprints/addons/ebs-csi-driver/) (_EbsCsiDriverAddOn_), to provide Persistent Volumes (PVs) fulfilling Konveyor's Persistent Volume Claims (PVCs)
+- a subdomain for the Konveyor application
+- a certificate for the subdomain, made available by the either _CreateCertificateProvider_ or _ImportCertificateProvider_, to be assigned to the load balancer
+-
```
Comment (jwmatthews) [inline docs/addons/konveyor.md:13]: Extra white space for outline to remove?

File: docs/addons/konveyor.md (hunk 5 above)
Comment (freschri) [reply docs/addons/konveyor.md:3]: done

File: lib/addons/olm/index.ts (hunk 2 above)
Comment (freschri) [reply lib/addons/olm/index.ts:39]: yes this is bad, addressed.

File: lib/addons/olm/index.ts (hunk 3 above)
Comment (freschri) [reply lib/addons/olm/index.ts:52]: addManifest finally the KubernetesManifest and it looks to me "manifest" is an array type in the param list, as per here: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_eks.KubernetesManifest.html#construct-props ?

File: lib/addons/olm/index.ts (hunk 4 above)
Comment (freschri) [reply lib/addons/olm/index.ts:56]: done

File: docs/addons/konveyor.md (hunk 7 above)
Comment (freschri) [reply docs/addons/konveyor.md:15]: done thanks

File: lib/addons/olm/index.ts (hunk 6 above)
Comment (freschri) [reply lib/addons/olm/index.ts:69]: done

File: lib/addons/olm/index.ts (hunk 3 above)
Comment (shapirov103) [reply lib/addons/olm/index.ts:56]: nm, I think we should do a better job on reflecting the types.

File: docs/addons/konveyor.md (hunk 5 above)
Comment (shapirov103) [reply docs/addons/konveyor.md:3]: Should this point to a different doc portal for tackle2 operator? The documentation page here is misleading and calls out the[ unmaintained repo](https://github.com/konveyor/konveyor.github.io/blob/main/content/_index.md)  as the source which is confusing (to me and I assume the target audience).

File: lib/addons/konveyor/tackle-manifest.ytpl (hunk 1 above)
Comment (shapirov103) [reply lib/addons/konveyor/tackle-manifest.ytpl:2]: Hello @jwmatthews, thank you for your feedback, it surely helps. Understand the comment related to the CR API version and upgrades. It is not as intuitive but it surely matches the saying "there is nothing more permanent than temporary" when it comes versioning. 

I assume we should change the source references to https://github.com/konveyor/tackle2-operator, please confirm. 

Is https://konveyor.github.io/ still the right documentation entry point? I saw a reference to tackle document [here](https://github.com/konveyor/tackle2-operator) pointing to what appears to be the right page: https://konveyor.github.io/tackle. This url, however, is not functioning. 

Understand that doc maintenance may not be the top priority, I was a lot more interesting in dev activities, which you confirmed. Thank you.

**Review by shapirov103 — COMMENTED** [review]
Great progress, a couple of mostly minor notes.

File: docs/addons/konveyor.md (hunk 5 above)
Comment (jwmatthews) [reply docs/addons/konveyor.md:3]: https://konveyor.github.io/ is good to use for Konveyor documentation.   The repo was stale and needed attention, hence we updated it ~2 weeks ago and are working to revamp this as the central documentation for Konveyor going forward.

File: lib/addons/konveyor/tackle-manifest.ytpl (hunk 1 above)
Comment (jwmatthews) [reply lib/addons/konveyor/tackle-manifest.ytpl:2]: Yes, https://konveyor.github.io/ is the right documentation entry point, we are bringing back usage of this as the central spot for developers to contribute to.

The source code for the Konveyor operator is located at: https://github.com/konveyor/tackle2-operator

If you want to reference source code, use: https://github.com/konveyor/tackle2-operator
If you want to reference documentation use: https://konveyor.github.io/ 

As to https://konveyor.github.io/tackle, that is an older link and not intended.

**Comment by freschri** [comment]
I am closing this PR, following recommendation from Mikhail, requested Claranet availability to host the 2 addons, they confirmed:
repo: https://github.com/claranet-ch/konveyor-eks-blueprint-addon
npm registry: https://www.npmjs.com/package/@claranet-ch/konveyor-eks-blueprint-addon

**Pull request closed without merging**